app.py

```diff
@@ -162,17 +177,61 @@ def show_weights(self):
         if len(self.weights) == 0:
             return
         if self.show_weight.get():
-            for weight in self.weights:
-                x, y, value = weight
+            for line in self.canvas_obj[self.OBJ_TYPE.LINE].values():
+                x_start,y_start = self.coordinates[line['point_start']]
+                x_end, y_end = self.coordinates[line['point_end']]
+                x, y = self.medium_coordinates(x_start,y_start,x_end, y_end)
+                value = line['weight']
                 r = int(self.r / 2) * len(str(value))
-                self.ids.append(self.canvas.create_oval(x - r, y - r, x + r, y + r, fill=self.BG, width=0))
-                self.ids.append(self.canvas.create_text(x, y, text=value, font="{} {}".format(self.FONT, str(r))))
+                weight_id = self.canvas.create_oval(x - r, y - r, x + r, y + r, fill=self.BG, width=0)
+                text_id = self.canvas.create_text(x, y, text=value, font="{} {}".format(self.FONT, str(r)))
+                self.canvas_obj[self.OBJ_TYPE.WEIGHT][weight_id] = {'text_id':text_id}
+                line['weight_id'] = weight_id
         else:
-            for element_id in self.ids:
-                self.canvas.delete(element_id)
+            for key, values in self.canvas_obj[self.OBJ_TYPE.WEIGHT].iteritems():
+                self.canvas.delete(key)
+                self.canvas.delete(values['text_id'])
+
+    def medium_coordinates(self,x_start,y_start,x_stop,y_stop):
+        return (x_start + x_stop) / 2, (y_start + y_stop) / 2
+
+    def movePoint(self, event):
+        # move the point and edges
+
+        x = self.canvas.canvasx(event.x)
+        y = self.canvas.canvasy(event.y)
+        obj_id = self.canvas.find_overlapping(x - 5, y - 5, x + 5, y + 5)
+        for obj in obj_id:
+            if obj in self.canvas_obj[self.OBJ_TYPE.POINT].keys():
+                obj_number = self.canvas_obj[self.OBJ_TYPE.POINT][obj]['idx']
+                self.canvas.coords(obj, event.x - self.r, event.y - self.r, event.x + self.r, event.y + self.r)
+                self.canvas.coords((self.objDict[obj][0]), event.x, event.y)
```

Comment:
How can you access self.objDict if you even don't have it? Did you try to run the code before commit? This line seems to throw error every time. Do you need it?